### PR TITLE
FH-4398 - Dependency Security Vulnerabilities

### PR DESCRIPTION
**Pull request by camilamacedo86**
## JIRA:
Link : https://issues.jboss.org/browse/FH-4398

## WHAT:
Fix High and Medium Security Vulnerabilities. 

## HOW:
* Upgrade the dependences browserify@3.46.1 and browserify-shim@3.3.2.
* Perform changes in the Gruntfile.js regards the breakchanges of this libs
* Add the dev dep `"phantomjs": "1.9.7-15"` to keep it compatible with NodeJS 4.
* Upgrade the dev dependece `grunt-browserify` for keep browserify grunt tasks running with success to test the project. 

### Changelogs of the dependencies upgraded:
* browserify : https://github.com/browserify/browserify/blob/master/changelog.markdown
* grunt-browserify : https://github.com/jmreidy/grunt-browserify/blob/master/CHANGELOG.md
* browserify-shim: https://gemnasium.com/npms/browserify-shim

## WHY
To solve the following vulnerabilities.
* Command Injection 
* Improper minification of non-boolean comparisons
* Insecure Randomness
* Regular Expression Denial of Service (DoS)
* Regular Expression Denial of Service (DoS)

### Production:
To minify the files/code. It is used in the following task which is executed when the lib is installed (`npm install`). 

```
Running "browserify:dist" (browserify) task
found current version = 3.0.3
Version to inject is 3.0.3
>> Bundle dist/feedhenry.js created.
``` 

### Development:
Used to do the dist and in the grunt task executed to test the lib. (`$grunt`)

```
Running "browserify:dist" (browserify) task
found current version = 3.0.3
Version to inject is 3.0.3
>> Bundle dist/feedhenry.js created.

Running "browserify:require" (browserify) task
>> Bundle test/browser/feedhenry-latest-require.js created.

Running "browserify:test" (browserify) task
>> Bundle ./test/browser/browserified_tests.js created.
```

## Steps to test:
Check if the CI/Process finilized with succes, since it will install and run all grunt taks for the lib with the all supported NodeJS versions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,6 +95,7 @@
     "json2lcov": "^1.0.0",
     "mocha": "1.17.1",
     "moment": "^2.18.1",
+    "phantomjs": "1.9.7-15",
```
Comment (davidffrench) [inline package.json]: What is the reason phantomjs is now needed as a dependency?

Comment (camilamacedo86) [reply package.json]: To keep working in NodeJS 4.

Comment (camilamacedo86) [reply package.json]: @davidffrench folllowing the answers. 

1. Do you know where browserify is used in this sdk? With such a large bump in the major version, I would be conscious about breaking changes.

The browsefly is used to minify the files which is the following task when npm install is executed. 

```
Running "browserify:dist" (browserify) task
found current version = 3.0.3
Version to inject is 3.0.3
>> Bundle dist/feedhenry.js created.
```
2. There should be license changes with these dependency updates also.
I will check it.

Comment (camilamacedo86) [reply package.json]: @davidffrench  I updated the first comment with all answers.

Comment (davidffrench) [reply package.json]: Perfect, sounds good to me @camilamacedo86

**Comment by camilamacedo86** [comment]
@davidffrench this project is not setup properly to update the licenses. The scripts were executed manually and a task was created in order to we fix that: https://issues.jboss.org/browse/RHMAP-20434.

**Pull request merged**